src/frontend/src/featureFlags/index.ts

```diff
@@ -0,0 +1,62 @@
+// Feature flags with default values
+const FEATURE_FLAGS_WITH_DEFAULTS = {
+  DOMAIN_COMPATIBILITY: false,
+} as const satisfies Record<string, boolean>;
+
+const LOCALSTORAGE_FEATURE_FLAGS_PREFIX = "ii-localstorage-feature-flags__";
+
+export class FeatureFlag {
+  readonly #storage: Pick<Storage, "getItem" | "setItem" | "removeItem">;
+  readonly #key: string;
+  readonly #defaultValue: boolean;
+  #value: boolean;
+
+  constructor(
+    storage: Pick<Storage, "getItem" | "setItem" | "removeItem">,
+    key: string,
+    defaultValue: boolean
+  ) {
+    this.#storage = storage;
+    this.#key = key;
+    this.#defaultValue = defaultValue;
+    const storedValue = this.#storage.getItem(this.#key);
+    this.#value =
```

Comment:
When we enable a flag by changing the default value, the users that disabled it, won't see it.

Is this what we want?

Shouldn't setting the value to `true` as default set it to `true` in all users?

Reply:
I think for (testing) purposes (we or users) might still want to disable a feature that's enabled by default at some point. By having the default value always enforced when it's true, you can't disable it anymore.

If we want all users to have a feature enabled without a possibility to disable it, the flag should be removed instead.

Reply:
We consider this as a testing only feature then, not to enable / disable features for a/b testing or to do partial roll outs of features.

Makes sense to me.